Doc/library/compression.zstd.rst

```diff
@@ -0,0 +1,770 @@
+   the :class:`ZstdFile` is closed.
+
+   The *mode* argument can be either ``'r'`` for reading (default), ``'w'`` for
+   overwriting, ``'x'`` for exclusive creation, or ``'a'`` for appending. These
+   can equivalently be given as ``'rb'``, ``'wb'``, ``'xb'`` and ``'ab'``
+   respectively.
+
+   If *file* is a file object (rather than an actual file name), a mode of
+   ``'w'`` does not truncate the file, and is instead equivalent to ``'a'``.
+
+   When opening a file for reading, the *options* argument can be a dictionary
+   providing advanced decompression parameters, see
+   :class:`DecompressionParameter` for detailed information about supported
+   parameters. The *zstd_dict* argument is a :class:`!ZstdDict` instance to be
+   used during decompression. When opening a file for reading, if the *level*
+   argument is passed a :exc:`!TypeError` will be raised.
+
+   When opening a file for writing, the *options* argument can be a dictionary
+   providing advanced decompression parameters, see
+   :class:`CompressionParameter` for detailed information about supported
+   parameters. The *level* argument is the compression level to use when
+   writing compressed data. Only one of *level* or *options* may be passed. The
+   *zstd_dict* argument is a :class:`!ZstdDict` instance to be used during
+   compression.
+
+   :class:`!ZstdFile` supports all the members specified by
+   :class:`io.BufferedIOBase`, except for :meth:`~io.BufferedIOBase.detach`
+   and :meth:`~io.IOBase.truncate`.
+   Iteration and the :keyword:`with` statement are supported.
+
+   The following method and attributes are also provided:
+
+   .. method:: peek(size=-1)
+
+      Return buffered data without advancing the file position. At least one
+      byte of data will be returned, unless EOF has been reached. The exact
+      number of bytes returned is unspecified (the *size* argument is ignored).
+
+      .. note:: While calling :meth:`peek` does not change the file position of
+         the :class:`ZstdFile`, it may change the position of the underlying
+         file object (for example, if the :class:`ZstdFile` was constructed by
+         passing a file object for *file*).
+
+   .. attribute:: mode
+
+      ``'rb'`` for reading and ``'wb'`` for writing.
+
+   .. attribute:: name
+
+      The name of the Zstandard file. Equivalent to the :attr:`~io.FileIO.name`
+      attribute of the underlying :term:`file object`.
+
+
+Compressing and decompressing data in memory
+--------------------------------------------
+
+.. function:: compress(data, level=None, options=None, zstd_dict=None)
+
+   Compress *data* (a :term:`bytes-like object`), returning the compressed
+   data as a :class:`bytes` object.
+
+   The *level* argument is an int object controlling the level of
+   compression. Refer to :meth:`CompressionParameter.bounds` to get the
+   values that can be passed for *level*. If advanced compression options are
```

Comment:
Clarification: the upper and lower bounds are specific to that `CompressionParameter` instance? Is there a default range one is most likely to expect?

Reply:
> Worth making explicit?

Yeah, I'd like to do so but I don't think sphinx understands IntEnums so I can't refer to the bounds reference and say `CompressionParameter.compression_level.bounds()`. Perhaps I can say (paraphrasing) "use `CompressionParameter.compression_level.bounds()`; refer to `bounds()` (a reference link) for details."

Reply:
Something like “_level_ is an alternative to _CompressionParameter.compression_level_” (if actually true) would be a start. You could also say “call _bounds_ on _compression_level_ to get the range of values for _level_” if you want, but that doesn’t seem so important if the link between _level_ and _compression_level_ is already make clear.